ops/model.py

```diff
@@ -69,7 +71,11 @@
 _StorageDictType = Dict[str, Optional[List['Storage']]]
 _BindingDictType = Dict[Union[str, 'Relation'], 'Binding']
 
-_StatusDict = TypedDict('_StatusDict', {'status': str, 'message': str})
+_ReadOnlyStatusName = Literal['error', 'unknown']
+_SettableStatusName = Literal['active', 'blocked', 'maintenance', 'waiting']
+StatusName = Union[_SettableStatusName, _ReadOnlyStatusName]
```

Comment:
We need to add this to `ops/__init__.py`, so it's accessible from there and shows up in the [docs](https://ops--1354.org.readthedocs.build/en/1354/search.html?q=statusname&check_keywords=yes&area=default).